src/TestInfrastructure/Helpers/MefTestHelpers.cs

```diff
@@ -103,10 +107,13 @@ public static Export CreateExport<T>(object exportInstance, string contractName
             for (int i = 0; i < requiredExports.Count(); i++)
             {
                 // Try importing when not all of the required exports are available -> exception
-                Exceptions.Expect<Exception>(
-                    null,
-                    () => TryCompose(importer, catalog, requiredExports.Take(i)),
-                    checkDerived: true);
+                var subsetOfRequiredExports = new List<Export>(requiredExports);
+                var exportToRemove = subsetOfRequiredExports[i];
```

Comment:
This fixes the other error in `CheckTypeCanBeImported` i.e. the one described in #2467.